File: app/src/main/java/com/amandariu/tagger/demo/data/source/local/db/TagDao.java
package com.amandariu.tagger.demo.data.source.local.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

/**
 * Dao for working with the tags table.
 *
 * @author amandariu (11/12/17)
 */
@Dao
public interface TagDao {
    @Query("select * from tags")
    List<TagEntity> loadAllTags();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertAll(List<TagEntity> tags);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void save(TagEntity tag);
}

File: app/src/main/java/com/amandariu/tagger/demo/data/source/local/db/TagDao.kt
package com.amandariu.tagger.demo.data.source.local.db

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query

/**
 * Dao for working with the tags table.
 *
 * @author Amanda Riu
 */
@Dao
interface TagDao {
    @Query("select * from tags")
    fun loadAllTags(): List<TagEntity>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(tags: List<TagEntity>)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun save(tag: TagEntity)
}
